mathesar_ui/src/stores/tableData.ts

```diff
@@ -519,6 +521,27 @@ export function getTable(db: string, id: number, options?: Partial<TableOptionsD
   return table;
 }
 
+export async function addColumn(db:string, id:number, new_column:TableColumn): Promise<void> {
+  const table = databaseMap.get(db)?.get(id);
+  if (table) {
+    await postAPI<unknown>(`/tables/${id}/columns/`, new_column)
+      .then((column:TableColumn) => {
+        void fetchTableDetails(db, id);
+        void fetchTableRecords(db, id, true);
+        return column;
+      })
+      .catch((err) => {
+        const tableColumnStore = table.columns;
+        tableColumnStore.set({
+          state: States.Error,
+          error: err instanceof Error ? err.message : null,
+          data: [],
+          primaryKey: null,
+        });
+      });
```

Comment:
I do not understand the need of using `.then` and `.catch` with the `async await` syntax, unless I'm missing something.

This could be updated to:
```
try {
  const column = await postAPI<TableColumn>(`/tables/${id}/columns/`, new_column);
  void fetchTableDetails(db, id);
  void fetchTableRecords(db, id, true);
  ...
} catch(err) {
  // catch block
 ...
}
```

Reply:
Have adjusted as suggested. Thanks!

Reply:
https://github.com/pegnott/mathesar/blob/754414d202a29033bd437f2d845269d5885a5831/mathesar_ui/src/stores/tableData.ts#L536